lib/api/user_token_service.rb

```diff
@@ -3,21 +3,25 @@ class UserTokenService
     def initialize(config = Settings, args = {})
       @config = config
       @svc_options = args
-      @token_mgr = new_token_mgr(base_config[:module], base_config[:name], api_config)
     end
 
-    attr_accessor :token_mgr
+    def token_mgr(namespace = nil)
+      @token_mgr ||= {}
+      if namespace.nil? || namespace == 'ui' # The default token and UI token share the same TokenStore namespace
+        @token_mgr['ui'] ||= new_token_mgr(base_config[:module], base_config[:name], api_config)
```

Comment:
Not sure if ui namespace makes sense,  UI fetches API tokens (i.e. api namespace, they're really api tokens), they just have a longer ttl. look at API.login in app/assets/javascripts/miq_api.js.   I was expecting namespaces **api** (for default, api, requester-type=ui)  and **ws** (for requested-type=ws)